src/strategies/revest/index.ts

```diff
@@ -0,0 +1,158 @@
+];
+
+const abi3 = [
+  "function totalLPAllocPoint() external view returns (uint)",
+  "function totalBasicAllocPoint() external view returns (uint)"
+]
+
+export async function strategy(
+  _space,
+  network,
+  _provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const PRECISION = BigNumber.from('1000000000000000000000');
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const eip1155OwnersParams = {
+    accounts: {
+      __args: {
+        where: {
+          id_in: addresses.map((a) => a.toLowerCase())
+        }
+      },
+      id: true,
+      balances: {
+        value: true,
+        token: {
+          registry: {
+            id: true,
+          },
+          identifier: true
+        }
+      }
+    }
+  };
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    eip1155OwnersParams.accounts.__args.block = { number: snapshot };
+  }
+  try {
+    let idsToCheck = new Set<string>();
+
+    const result = await subgraphRequest(
+      SUBGRAPH_URL[network],
+      eip1155OwnersParams
+    );
+    result.accounts.forEach(element => {
+      element.relBal = element.balances.filter((balance) => {
+        const isRightAddress =
+          balance.token.registry.id.toLowerCase() === options.address.toLowerCase() && balance.value != '0';
+        if(isRightAddress) {
+          idsToCheck.add(balance.token.identifier);
+        }
+        return isRightAddress;
+      });
+    });
+    let ids = Array.from(idsToCheck);
+    const response = await multicall(
+      network,
+      _provider,
+      abi2,
+      ids.map((id: any) => [options.staking, 'getDisplayValues', [id, 0]]),
+      {blockTag}
+    );
+
+    let fnfts = Object.fromEntries(
+      response.map((value, i) => [
+        ids[i],
+        defaultAbiCoder.decode(['uint', 'uint'], value[0])[0]
+    ]));
+    Object.keys(fnfts).forEach(element => {
+      if(fnfts[element].eq('0')) {
+        delete fnfts[element];
+      }
+    })
+    ids = Object.keys(fnfts);
+    const response2 = await multicall(
+      network,
+      _provider,
+      abi,
+      ids.map((id: any) => [options.tokenVault, 'getFNFT', [id]]),
+      {blockTag}
+    );
+    let completeFNFTs = Object.fromEntries(
+      response2.map((value, i) => [
+        ids[i],
+        {allocPoints: fnfts[ids[i]], isRVST: value[0].asset.toLowerCase() == options.token.toLowerCase()}
+    ]));
+    let rewards = await multicall(
+      network,
+      _provider,
+      abi2,
+      [""].map((id: any) => [options.staking, 'rewardsHandlerAddress', []]),
+      {blockTag}
+    );
+    rewards = rewards[0][0];
+    let allocLP = await multicall(
+      network,
+      _provider,
+      abi3,
+      [""].map((id: any) => [rewards, 'totalLPAllocPoint', []]),
+      {blockTag}
+    );
+    let allocToken = await multicall(
```

Comment:
Hi @RobAnon There is a limit on the number of requests being sent (multicall/subgraph/fetch), It is limited to 5 calls 

https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Can you try to find if there is some alternate way to get this in fewer calls? maybe try sending multiple calls into multicall

Reply:
Can I call multiple functions via multicall?

Reply:
@ChaituVR I've reduced the number to 5 calls through the use of multicall